core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableSchema.java

```diff
@@ -23,6 +23,12 @@
     private final List<ColumnSchema> columns;
     private final List<String> checkConstraints;
 
+    @Deprecated
```

Comment:
Let's make the new one experimental and remove deprecation. For now. We will add deprecation when the new one is no longer experimental.